client/store/modules/repositories/mutations.js

```diff
@@ -57,6 +58,9 @@ const addTag = (state, { tag, repositoryId }) => {
 const removeTag = (state, { tagId, repositoryId }) => {
   const repository = find(state.items, { id: repositoryId });
   repository.tags = repository.tags.filter(it => it.id !== tagId);
+  if (!state.tagFilter.length) return;
```

Comment:
You can move this on line 59, to return early as possible, if conditions are not met.

Reply:
If I move this check on the top it will stop execution of removing repository tags when there are no tag filters applied. This logic has been added to handle special use case